packages/asset-swapper/src/utils/market_operation_utils/multihop_utils.ts

```diff
@@ -0,0 +1,45 @@
+import { BigNumber } from '@0x/utils';
+import * as _ from 'lodash';
+
+import { ZERO_AMOUNT } from './constants';
+import { getTwoHopAdjustedRate } from './fills';
+import { DexSample, FeeSchedule, MarketSideLiquidity, MultiHopFillData, TokenAdjacencyGraph } from './types';
+
+/**
+ * Given a token pair, returns the intermediate tokens to consider for two-hop routes.
+ */
+export function getIntermediateTokens(
+    makerToken: string,
+    takerToken: string,
+    tokenAdjacencyGraph: TokenAdjacencyGraph,
+    wethAddress: string,
+): string[] {
+    let intermediateTokens = [];
+    if (makerToken === wethAddress) {
+        intermediateTokens = _.get(tokenAdjacencyGraph, takerToken, [] as string[]);
+    } else if (takerToken === wethAddress) {
+        intermediateTokens = _.get(tokenAdjacencyGraph, makerToken, [] as string[]);
+    } else {
+        intermediateTokens = _.union(
+            _.intersection(_.get(tokenAdjacencyGraph, takerToken, []), _.get(tokenAdjacencyGraph, makerToken, [])),
+            [wethAddress],
+        );
+    }
+    return intermediateTokens;
+}
+
+/**
+ * Returns the best two-hop quote and the fee-adjusted rate of that quote.
+ */
+export function getBestTwoHopQuote(
+    marketSideLiquidity: MarketSideLiquidity,
+    feeSchedule?: FeeSchedule,
+): { quote: DexSample<MultiHopFillData> | undefined; rate: BigNumber } {
```

Comment:
```suggestion
): { quote: DexSample<MultiHopFillData> | undefined; adjustedRate: BigNumber } {
```
just to avoid any potential bugs in assuming it is the rate rate.